src/common/components/feed/feed.utils.ts

```diff
@@ -138,19 +147,83 @@ export function generateFeedCardAttributes(
   }
 }
 
-export function shouldAggregate(e1: FeedQueryEvent, e2: FeedQueryEvent, e3: FeedQueryEvent) {
+export const getAggregateProfiles = (aggregateEvent: AggregateEvent): User[] => {
+  if (aggregateEvent.eventsAggregated.length < 2) {
+    return [
+      {
+        address: aggregateEvent.walletAddress,
+        profile: aggregateEvent.profile,
+      },
+    ];
+  }
+  const users: User[] = [];
+  aggregateEvent.eventsAggregated.map((user) => {
+    if (!users?.find((u) => u?.address === user.walletAddress)) {
+      users.push({
+        address: user.walletAddress,
+        profile: user.profile,
+      });
+    } else {
+      // do nothing
+    }
+  });
+  return users as User[];
+};
+
+export const aggregateEventsTime = (events: FeedQueryEvent[]) => {
```

Comment:
The choice of averaging the event times seems a little odd to me. If they are all very close in time then it doesnt make much difference, but if they are several minutes apart, I would opt for taking the earliest time to indicate "This set of events started at this time".  Kinda like if you went to a concert and averaged the start time of the cover band with the main show, versus saying the show started when the cover band did.